ImageStacks/ImageStacks.py

```diff
@@ -624,36 +639,65 @@ def loadVolume(self, outputNode=None):
       reader.SetFileName(path)
       image = reader.Execute()
       sliceArray = sitk.GetArrayFromImage(image)
-      if len(sliceArray.shape) == 3:
+      if len(sliceArray.shape) == 3 and self.outputGrayscale:
+        # We convert to grayscale by simply taking the first component, which is appropriate for cases when grayscale image is stored as R=G=B,
+        # but to convert real RGB images it could better to compute the mean or luminance.
         sliceArray = sitk.GetArrayFromImage(image)[:,:,0]
```

Comment:
We very rarely see real RGB volumes but if this comes up in practice people could use the VectorToScalarVolume module to get the luminance volume from a loaded RGB.

Reply:
We are likely to see more going forward, as our lab will likely to start working on cellular images soonish....

Reply:
Would be easy to add the formula (Y=0.2126R+0.7152G+0.0722B).  I don't think it would add much overhead and would be a no-op if r==g==b.

Reply:
Or add a selector of which channel to import.

If you are doing microscopy you should be aware of these two efforts, which have grown independently and may merge at some point:

This is just new from ProjectWeek: https://github.com/slicermicro

This has been brewing for a while but just came out in this form: https://slicerscope.github.io/

Plus we are starting to work with microscopy data in the Imaging Data Commons so probably we'll be using Slicer for part of that.

Reply:
This update is mainly for loading moderately lathe RGB volumes, such as the visible human cryosection photos.

Microscopy images would need slightly different processing, for example tiled loading (never load an entire plate into memory) and higher quality downsampling.